packages/suite/src/views/firmware/FirmwareModal.tsx

```diff
@@ -77,19 +82,13 @@ export const FirmwareModal = ({
             return <PrerequisitesGuide />;
         }
 
-        // Special and hopefully very rare case. This appears when somebody tried to fool user into using a hacked firmware. This check is skipped when installing custom FW.
-        if (device?.id && firmwareHashInvalid.includes(device.id)) {
-            return (
-                <OnboardingStepBox
-                    image="UNI_ERROR"
-                    heading={<Translation id="TR_FIRMWARE_HASH_MISMATCH" />}
-                    nested
-                />
-            );
-        }
-
         switch (status) {
             case 'error':
+                // Special and hopefully very rare case. This appears when somebody tried to fool user into using a hacked firmware. This check is skipped when installing custom FW.
+                if (device?.id && firmwareHashInvalid.includes(device.id)) {
+                    return <FirmwareUpdateHashCheckError error={error} />;
+                }
+
```

Comment:
Moving the code here is what enables you to reopen FW update modal, so you can try reinstalling it.

When the condition was above the switch, it meant that once a device was marked as failed, reopening the FW update modal would immediately display the error as if it just finished in error state, and never let you reinstall.
I assume that's a bug so this fixes it :shrug: